src/Model/Config/Communication.php

```diff
@@ -123,4 +102,49 @@ protected function getChannel(string $langAbbr): string
 
         return $channels[$langAbbr];
     }
+
+    private function prepareParams(array $params, Category $category = null): array
+    {
+        $params['add-params'] = $this->useForCategories() ? $this->getStandardCategoryPath($category) : 'cl=search_result';
+
+        if ($this->getConfig('ffApiVersion') === 'ng') {
+            $params['category-page'] = $category ?
+                $this->useForCategories() ? $this->getNgCategoryPath($category) : null
+                : null;
+            $params['add-params']    = null;
+        }
+
+        return $params;
+    }
+
+    private function getNgCategoryPath(Category $category, string $param = 'CategoryPath'): string
+    {
+        $categories = [rawurlencode($this->filter->filterValue($category->getTitle()))];
+
+        while ($parent = $category->getParentCategory()) {
+            $categories[] = rawurlencode($this->filter->filterValue($parent->getTitle()));
+            $category     = $parent;
+        }
+
+        $path = implode('/', array_reverse($categories));
+
+        return sprintf('navigation=true,filter=%s:%s', urlencode($param), urlencode($path));
```

Comment:
this will result with 
![image](https://user-images.githubusercontent.com/4511883/151961702-0d361acb-3a57-4332-864e-23112563c13a.png)
which is not a correct setup. It will not break a page or cause FACT-Finder to return wrong records but still we should know how to use our own tools

Reply:
let's encode `:` too like we have in other SDKs. According to documentation this is the recommended way